### ISSUE TITLE
S3925: .NET 8 warnings

### ISSUE THREAD

**Issue by IeuanWalker** [SonarSource/sonar-dotnet#8065]
I have this custom exception - 
```csharp
[Serializable]
public class CalendarDateChangedException : Exception
{
	public CalendarDateChangedException()
	{
	}

	public CalendarDateChangedException(string? message) : base(message)
	{
	}

	public CalendarDateChangedException(string? message, Exception? innerException) : base(message, innerException)
	{
	}

	protected CalendarDateChangedException(SerializationInfo info, StreamingContext context) : base(info, context)
	{
	}
}
```

Retarged the application from .net 7 to 8, i get this warning -
```
Warning	SYSLIB0051	'Exception.Exception(SerializationInfo, StreamingContext)' is obsolete: 'This API supports obsolete formatter-based serialization. It should not be called or extended by application code.' (https://aka.ms/dotnet-warnings/SYSLIB0051)	
```

Removing the last method, i then get this warning -
```
Warning	S3925	Update this implementation of 'ISerializable' to conform to the recommended serialization pattern.
```

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @IeuanWalker! 

I think the warning that you are experiencing is caused by [this .Net 8 breaking change](https://github.com/dotnet/docs/issues/34893). Do you need your custom exception type to be serializable?
If not, you can follow the recommended action (see "If you're creating a custom System.Exception-derived type" section), and remove the `[Serializable]` attribute, this way S3925 should not raise.

**Comment by Wouter29** [comment]
This is not working @cristian-ambrosini-sonarsource 
If I make a custom Exception like this
```
    public class CustomException : Exception
    {
    }
```
It still triggers a "S3925:\"ISerializable\" should be implemented correctly" that only can be solved by adding a suppression.

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @Wouter29! We are starting a dedicated effort over the next 2 weeks to address serialization rules post-Net8 (see #8377). Please follow the mentioned issue to stay updated on the progress.

cheers!

**Comment by martin-strecker-sonarsource** [comment]
The `[Serializable]` attribute applied in the code example is an indication, that binary formatting support is intended. Therefore S3925 is raised if the serialization constructor is missing.
You need to remove the `[Serializable]` attribute and the serialization constructor from the type.
See also https://github.com/SonarSource/sonar-dotnet/pull/7673

**Comment by AlexEngblom** [comment]
Cannot get rid of warning S3925 with these recommendations (also documented here https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051).
